File: ts-sdk/src/sdk/models/operations/deleteshieldinformationbarriersegmentsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteShieldInformationBarrierSegmentsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=shield_information_barrier_segment_id" })
  shieldInformationBarrierSegmentId: string;
}


export class DeleteShieldInformationBarrierSegmentsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: DeleteShieldInformationBarrierSegmentsIdPathParams;
}


export class DeleteShieldInformationBarrierSegmentsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/deleteshieldinformationbarriersegmentsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type DeleteShieldInformationBarrierSegmentsIDPathParams struct {
	ShieldInformationBarrierSegmentID string `pathParam:"style=simple,explode=false,name=shield_information_barrier_segment_id"`
}

type DeleteShieldInformationBarrierSegmentsIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams DeleteShieldInformationBarrierSegmentsIDPathParams
}

type DeleteShieldInformationBarrierSegmentsIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}
